hack/update-go-module-dependencies.sh

```diff
@@ -25,6 +25,9 @@
 # we also must take care of replacing modules such as sigs.k8s.io/cluster-api-provider-azure
 # with github.com/openshift/cluster-api-provider-azure (just an example, there are more).
 
+RELEASE=release-4.14
+GO_VERSION=1.20
```

Comment:
We will also need to update the Dockerfile.aro file used by the build pipeline. We don't need to add a new 1.20 image to ARO.Build.Images, since that was done in this [PR for 1.20.10](https://msazure.visualstudio.com/AzureRedHatOpenShift/_git/ARO.Build.Images/pullrequest/9512891).

Apart from that, once we merge this into release-4.14, I believe we'll need a separate PR for CI, similar to [PR 41496](https://github.com/openshift/release/pull/41496).